File: src/main/java/it/giaquinto/springberry/controller/system/CloseSystemController.java
package it.giaquinto.springberry.controller.system;

import it.giaquinto.springberry.controller.SpringBerrySystemController;
import it.giaquinto.springberry.model.system.SystemExit;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class CloseSystemController extends SpringBerrySystemController {

    @GetMapping(path = "/system/close")
    public void systemReboot() {
        System.exit(SystemExit.valueOf(SystemExit.TRIGGERED));
    }
}

File: src/main/java/it/giaquinto/springberry/controller/system/CloseSystemController.kt
package it.giaquinto.springberry.controller.system

import it.giaquinto.springberry.controller.SpringBerrySystemController
import it.giaquinto.springberry.model.system.SystemExit
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import kotlin.system.exitProcess

@RestController
class CloseSystemController : SpringBerrySystemController() {
    @GetMapping(path = ["/system/close"])
    fun systemReboot() {
        exitProcess(SystemExit.valueOf(SystemExit.TRIGGERED))
    }
}